staging/src/k8s.io/kubectl/pkg/cmd/set/set_env.go

```diff
@@ -45,7 +45,7 @@ import (
 var (
 	validEnvNameRegexp = regexp.MustCompile("[^a-zA-Z0-9_]")
 	envResources       = `
-  	pod (po), replicationcontroller (rc), deployment (deploy), daemonset (ds), job, replicaset (rs)`
+  	pod (po), replicationcontroller (rc), deployment (deploy), daemonset (ds), job, cronjob (cj), replicaset (rs)`
```

Comment:
Looks fine, but just a question.  What about `statefulset`?  Wouldn't that also be valid here?

Reply:
Thanks for pointing it out. StatefulSet is also valid here. (The funny thing is it is tested in the [set_env_test.go](https://github.com/kubernetes/kubernetes/blob/52eea971c57580c6b1b74f0a12bf9cc6083a4d6b/staging/src/k8s.io/kubectl/pkg/cmd/set/set_env_test.go#L375) but not described here.)

I also realized that job should not be here. Similar to the [issue](https://github.com/kubernetes/kubernetes/issues/48388) discussed here, env is also not updatable.

I’ll add a few more commits to this PR and update PR description.